httpclient5-testing/src/main/java/org/apache/hc/client5/testing/auth/Authenticator.java

```diff
@@ -31,6 +31,13 @@
 
 public interface Authenticator {
 
+    /**
+     * @since 5.3
+     */
+    default AuthResult auth(URIAuthority authority, String requestUri, String credentials) {
```

Comment:
Maybe `perform` is a better name? `auth` and `authenticate` seem to be to similar.

Reply:
@michael-o Agreed.